src/ui/components/NetworkMonitor/RequestDetails.tsx

```diff
@@ -309,7 +309,7 @@ const RequestDetails = ({ cx, request }: { cx: any; request: RequestSummary }) =
     }
   }, [activeTab, activeTabs]);
 
-  if (!(loadedRegions && request && getPointIsInLoadedRegion(loadedRegions, request.point.point))) {
+  if (!(loadedRegions && request && isPointInRegions(loadedRegions, request.point.point))) {
```

Comment:
Do we still need this code? It seems like we should not be able to trigger this anymore, or if we did, returning `null` might be ok?

Reply:
I added logic to prevent selecting a resource that's already outside of the loaded regions, but I didn't add anything to deselect a previously selected resource when a region gets unloaded.

We should probably handle that too, but currently we'll retain the state so that if you reload the region– the item is "reselected" automatically. In the interim, (when you've expected the trim but not yet loaded) we'll render the unloaded message.

If you'd like we can chat about the UX here. I'm not trying to fix all related issues with this PR FWIW though, just one of the more egregious ones I noticed personally.

Reply:
Goootchya. That makes sense, I have one further suggestion:

It sounds like now the *only* situation in which we would see this RequestUnavailable component is when the currently select request is a region that is not loaded. In that case, would we change that message from "this is unavailable" (which feels like it does not give the user a place to go any further into understanding what is going on) to "This request happened in a part of the recording which is not currently loaded"? That way it at least gives them something more specific to look for information about like "Why did part of my recording unload?"

Reply:
> This request happened in a part of the recording which is not currently loaded

Yeah, that sounds like a better message. I'll make that change 👍🏼